src/cc_helpers.cc

```diff
@@ -1167,49 +1168,32 @@ void print_org_name(X509_NAME* name) {
   printf("\n");
 }
 
-void print_ssl_error(int code) {
-  switch(code) {
-  case SSL_ERROR_NONE:
-    printf("No ssl error\n");
-    break;
-  case SSL_ERROR_WANT_READ:
-    printf("want read ssl error\n");
-    break;
-  case SSL_ERROR_WANT_WRITE:
-    printf("want write ssl error\n");
-    break;
-#ifndef BORING_SSL
-  case SSL_ERROR_WANT_CONNECT:
-    printf("want connect ssl error\n");
-    break;
-#endif
-  case SSL_ERROR_WANT_ACCEPT:
-    printf("want accept ssl error\n");
-    break;
-  case SSL_ERROR_WANT_X509_LOOKUP:
-    printf("want lookup ssl error\n");
-    break;
+/* Declare an array of SSL Error codes mapped to an error description */
+optlookup Ssl_errors[] =
+  {
+      DCL_OPTLOOKUP(SSL_ERROR_NONE                 , "No ssl error")
+    , DCL_OPTLOOKUP(SSL_ERROR_WANT_READ            , "want read error")
+    , DCL_OPTLOOKUP(SSL_ERROR_WANT_WRITE           , "want write error")
 #ifndef BORING_SSL
-  case SSL_ERROR_WANT_ASYNC:
-    printf("want async ssl error\n");
-    break;
-  case SSL_ERROR_WANT_CLIENT_HELLO_CB:
-    printf("wantclient hello  ssl error\n");
-    break;
-#endif
-  case SSL_ERROR_SSL:
-    printf("ssl error error\n");
-    break;
-  case SSL_ERROR_SYSCALL:
-    printf("ssl error syscall\n");
-    break;
-  case SSL_ERROR_ZERO_RETURN:
-    printf("ssl error zero return\n");
-    break;
-  default:
-    printf("Unknown ssl error, %d\n", code);
-    break;
-  }
+    , DCL_OPTLOOKUP(SSL_ERROR_WANT_CONNECT         , "want connect error")
+    , DCL_OPTLOOKUP(SSL_ERROR_WANT_ASYNC           , "want async error")
+    , DCL_OPTLOOKUP(SSL_ERROR_WANT_CLIENT_HELLO_CB , "want client hello error")
+#endif   // BORING_SSL
+    , DCL_OPTLOOKUP(SSL_ERROR_WANT_ACCEPT          , "want accept error")
+    , DCL_OPTLOOKUP(SSL_ERROR_WANT_X509_LOOKUP     , "want X509 lookup error")
+    , DCL_OPTLOOKUP(SSL_ERROR_SSL                  , "generic ssl error")
+    , DCL_OPTLOOKUP(SSL_ERROR_SYSCALL              , "syscall error")
+    , DCL_OPTLOOKUP(SSL_ERROR_ZERO_RETURN          , "zero return error")
+  };
+
+// Method to convert SSL error-code to human-readable string, a la' strerror()
+static inline const char * ssl_strerror(int code) {
+  const char *msg = optbyid(Ssl_errors, code);
+  return(msg ? msg : "Unknown ssl error, " CC_TO_STR(code));
+}
+
+void print_ssl_error(int code) {
+  printf("%s\n", ssl_strerror(code));
```

Comment:
Bulk of the old code in this print method greatly simplified by just doing a lookup from the lookup array.